packages/gatsby-plugin-image/src/components/hooks.ts

```diff
@@ -38,18 +38,37 @@ export function storeImageloaded(cacheKey?: string): void {
 export function hasImageLoaded(cacheKey: string): boolean {
   return imageCache.has(cacheKey)
 }
-
+export type IGatsbyImageDataParent<T = never> = T & {
+  gatsbyImageData: IGatsbyImageData
+}
 export type FileNode = Node & {
-  childImageSharp?: Node & {
-    gatsbyImageData?: IGatsbyImageData
+  childImageSharp?: IGatsbyImageDataParent<Node>
+}
+
+export const isGatsbyImageData = (
+  node: IGatsbyImageData | any
+): node is IGatsbyImageData => Boolean(node?.images?.fallback?.src) // 🦆
```

Comment:
Is this duck comment a question? Boolean is fine, an if statement might be slightly more readable.

Reply:
Oh, it's just saying that I'm duck typing here by checking for specific props. Clearly should have better comments!